lib/logstash/inputs/eventlog.rb

```diff
@@ -21,120 +22,62 @@ class LogStash::Inputs::EventLog < LogStash::Inputs::Base
-      }.each{
-          |property| e[property] = event.send property
-      }
-
-      if LogStash::Environment.jruby?
-        # unwrap jruby-win32ole racob data
-        e["InsertionStrings"] = unwrap_racob_variant_array(event.InsertionStrings)
-        data = unwrap_racob_variant_array(event.Data)
-        # Data is an array of signed shorts, so convert to bytes and pack a string
-        e["Data"] = data.map{|byte| (byte > 0) ? byte : 256 + byte}.pack("c*")
-      else
-        # win32-ole data does not need to be unwrapped
-        e["InsertionStrings"] = event.InsertionStrings
-        e["Data"] = event.Data
+      new_total = @eventlog.total_records()
+      if new_total != old_total
+        rec_num = @eventlog.oldest_record_number() if @eventlog.full?
+        @eventlog.read(flags, rec_num).each { |log| e = process(log); decorate(e); queue << e }
+        old_total = new_total
+        rec_num   = @eventlog.read_last_event.record_number + 1
       end
+      Stud.stoppable_sleep(@interval/1000.0) { stop? }
+    end
+  end
 
-      e["message"] = event.Message
-
-      decorate(e)
-      queue << e
-
-    end # loop
-
-  rescue LogStash::ShutdownSignal
-    return
-  rescue => ex
-    @logger.error("Windows Event Log error: #{ex}\n#{ex.backtrace}")
-    sleep 1
-    retry
-  end # def run
-
-  private
-  def unwrap_racob_variant_array(variants)
-    variants ||= []
-    variants.map {|v| (v.respond_to? :getValue) ? v.getValue : v}
-  end # def unwrap_racob_variant_array
-
-  # the event log timestamp is a utc string in the following format: yyyymmddHHMMSS.xxxxxx±UUU
-  # http://technet.microsoft.com/en-us/library/ee198928.aspx
   private
-  def to_timestamp(wmi_time)
-    result = ""
-    # parse the utc date string
-    /(?<w_date>\d{8})(?<w_time>\d{6})\.\d{6}(?<w_sign>[\+-])(?<w_diff>\d{3})/ =~ wmi_time
-    result = "#{w_date}T#{w_time}#{w_sign}"
-    # the offset is represented by the difference, in minutes,
-    # between the local time zone and Greenwich Mean Time (GMT).
-    if w_diff.to_i > 0
-      # calculate the timezone offset in hours and minutes
-      h_offset = w_diff.to_i / 60
-      m_offset = w_diff.to_i - (h_offset * 60)
-      result.concat("%02d%02d" % [h_offset, m_offset])
-    else
-      result.concat("0000")
-    end
+  def process(log)
+
+    LogStash::Event.new(
+      "host"             => @hostname,
+      "Logfile"          => @logfile,
+      "message"          => log["description"].strip,
+      "Category"         => log["category"],
+      "ComputerName"     => log["computer"],
+      "EventIdentifier"  => log["event_id"],
+      "EventType"        => log["event_type"],
+      "RecordNumber"     => log["record_number"],
+      "SourceName"       => log["source"],
+      "TimeGenerated"    => log["time_generated"],
+      "TimeWritten"      => log["time_written"],
+      "Type"             => log["event_type"],
+      "User"             => log["user"],
+      "InsertionStrings" => log["string_inserts"]
```

Comment:
Are all this fields being populated for every log line? Not saying that they wont, just curious..